src/core/crypto/hkdf_sha256.cpp

```diff
@@ -32,63 +32,92 @@
  */
 
 #include "hkdf_sha256.hpp"
+#include "common/code_utils.hpp"
+#include "common/error.hpp"
 
 #include <string.h>
 
 namespace ot {
 namespace Crypto {
 
-void HkdfSha256::Extract(const uint8_t *aSalt, uint16_t aSaltLength, const uint8_t *aInputKey, uint16_t aInputKeyLength)
+void HkdfSha256::Extract(const uint8_t *aSalt, uint16_t aSaltLength, otCryptoKey *aKey)
 {
-    HmacSha256 hmac;
+    
+    if(otPlatCryptoGetType() == OT_CRYPTO_TYPE_PSA)
+    {
+        Error err = otPlatCryptoHkdfExtract(&mOperation, aSalt, aSaltLength, aKey->mKeyRef);
+        (void)err;
+    }
+    else
+    {
+        HmacSha256 hmac;
+        otCryptoKey cryptoKey;
 
-    // PRK is calculated as HMAC-Hash(aSalt, aInputKey)
+        cryptoKey.mKey = aSalt;
+        cryptoKey.mKeyLength = aSaltLength;
+        cryptoKey.mKeyRef = 0;
 
-    hmac.Start(aSalt, aSaltLength);
-    hmac.Update(aInputKey, aInputKeyLength);
-    hmac.Finish(mPrk);
+        // PRK is calculated as HMAC-Hash(aSalt, aInputKey)
+        hmac.Start(&cryptoKey);
+        hmac.Update(aKey->mKey, aKey->mKeyLength);
+        hmac.Finish(mPrk);
```

Comment:
Yes, I have the default `otPlatCryptoHkdfExpand` implementation, which uses mbedtls,  in `crypto_platform.cpp` I think this can go into the `otPlatCryptoHkdfExpand` in there.